libfoundation/src/foundation-string.cpp

```diff
@@ -4841,7 +4841,25 @@ static bool __MCStringExpandAt(MCStringRef self, uindex_t p_at, uindex_t p_count
 	// Fetch the capacity.
 	uindex_t t_capacity;
 	t_capacity = __MCStringGetCapacity(self);
-    
+
+	/* Overflow check.  A string's length needs to fit into an
+	 * index_t, because sometimes negative indices are used.  Ensure that the
+	 * length of the expanded string (including a null byte) fits. */
+	if (INDEX_MAX == p_count || INDEX_MAX - p_count - 1 < self->char_count)
```

Comment:
@runrevmark  When I was checking against `UINDEX_MAX`, including enough margin for the nul byte ensured that it was safe to loop over the string pointer using a `uindex_t` offset (I believe there is currently code that does that in the engine).  I've kept the extra margin even with the `UINDEX_T` limit, because I don't want someone to come along later, change `INDEX_MAX` to `UINDEX_MAX` and accidentally acquire subtle bugs.